cmake/external/triton.cmake

```diff
@@ -44,13 +44,11 @@ if (WIN32)
   vcpkg_install(re2)
   vcpkg_install(boost-interprocess)
   vcpkg_install(boost-stacktrace)
-  vcpkg_install(zlib)
   vcpkg_install(pthread)
   vcpkg_install(b64)
 
   add_dependencies(getb64 getpthread)
-  add_dependencies(getpthread getzlib)
-  add_dependencies(getzlib getboost-stacktrace)
+  add_dependencies(getpthread getboost-stacktrace)
   add_dependencies(getboost-stacktrace getboost-interprocess)
   add_dependencies(getboost-interprocess getre2)
```

Comment:
We already have re2.  We cannot have two versions of the same lib.

Reply:
but triton won't build - it goes to vcpkg for dependencies, not us.

Reply:
Triton's http-client-library is relatively simple. It only has a few sources files and doesn't have many dependencies. You may write a CMakeList.txt for it if theirs doesn't meet our needs. For example, for ONNX we have https://github.com/microsoft/onnxruntime/blob/main/cmake/external/onnx_minimal.cmake . And if your need deeper at their source code, you might found it just does the same thing what you are doing for the OpenAI EP.  You might rewrite the code and extract the common part and reuse the common parts for the two different types of endpoints. 
Since your code needs to be statically linked to the other parts of ORT, all of them need to use the same way to manage dependencies. Either we all use vcpkg, or we all do not use vcpkg. We cannot just use vcpkg for some part of the code. It needs be consistent.

Reply:
For example, if you are writing a custom op, you are free to use anyway to get your dependencies.  Because custom ops live in a separated DLL.  We do not worry about the their versions as long as our custom op API has a stable binary interface(ABI). 
But your this code needs to be statically link to ORT, so it is different.